apps/alarm/alarm.html

```diff
@@ -0,0 +1,103 @@
+<html>
+  <head>
+    <link rel="stylesheet" href="../../css/spectre.min.css">
+  </head>
+  <body>
+
+    <p><button id="fetch" class="btn btn-primary">Fetch Language from Bangle.js</button></p>
+    <div id="formGroup" class="form-group" style="visibility: hidden;">
+    <p>Then please choose a language from the following list:</p>
+    <select id="languages" class="form-select">
+    </select>
+    </div>
+    <p><button id="upload" class="btn btn-primary" style="visibility: hidden;">Then Upload</button></p>
+
+    <script src="../../lib/customize.js"></script>
+    <script src="https://www.puck-js.com/puck.js"></script>
+
+    <script>
+      let request = url => new Promise( (resolve, reject) => {
+          let xhr = new XMLHttpRequest();
+          xhr.open("GET", url);
+          xhr.onload = () => {
+              if (xhr.status >= 200 && xhr.status < 300) {
+                  resolve(xhr.response);
+              } else {
+                  reject(xhr.statusText);
+              }
+          };
+          xhr.onerror = () => reject(xhr.statusText);
+          xhr.onabort = () => reject(xhr.statusText);
+          xhr.send();
+      });
+
+      let translated = {};
+	  let translate = () => {
+	    Promise.all([translateFile("app.js"), translateFile("alarm.js")]).then(() => {
+          document.getElementById("upload").style.visibility = 'visible';
+        }).catch( error => {
+          console.log(error);
+          document.getElementById("upload").style.visibility = 'hidden';
+        });
+	  };
+      let translateFile = (fileName) => {
+        var languageSelector = document.getElementById("languages");
+        var lang = languageSelector.options[languageSelector.selectedIndex].value;
+        console.log(lang);      
+        request(fileName).then( fileContent => 
+          request("app_" + lang + ".json")
+          .catch( () => request("app_" + lang.substring(0,2) + ".json"))
+          .catch( () => request("app.json"))
+          .then( langjson => {
+            var trans = JSON.parse(langjson);
+            fileContent = fileContent.replace(/"([^"]*?)"\/\*LANG\*\//g, function(m, p1) { return '"' + (trans[p1]||p1) + '"' + (trans[p1]?'':'/*LANG*/'); });
+            fileContent = fileContent.replace(/'([^']*?)'\/\*LANG\*\//g, function(m, p1) { return "'" + (trans[p1]||p1) + "'" + (trans[p1]?'':'/*LANG*/'); });
```

Comment:
this is clever, but I think it would be better to just use a function in the callsite, something like https://www.i18next.com/translation-function/essentials#passing-a-default-value (and many others) use:

`locale.t('new', 'New Alarm')`

Using the English (here) as the key is not good, if the English text “New Alarm” changes, then you can't reuse the translations properly.

Actually, there's already a locale.translate function: https://www.espruino.com/Bangle.js+Locale#translation
but it uses the English text, and doesn't support Application resources.

Needs some work!